### PR TITLE
ipv6_hdr: Initial import

### PR DESCRIPTION
**Pull request by authmillenon**
Taken out of #2454 to be used by other PRs that need this header (e.g. #2430).

Based on #2730 for checksum calculation (merged).

### CODE REVIEW

File: Makefile.dep [hunk 1]
```diff
@@ -48,6 +48,10 @@ ifneq (,$(filter sixlowpan,$(USEMODULE)))
   USEMODULE += vtimer
 endif
 
+ifneq (,$(filter ng_ipv6_hdr,$(USEMODULE)))
+  USEMODULE += ng_inet_csum
```
Comment (Kijewski) [inline Makefile.dep]: Totally unrelated to this PR: The prefix NG sucks. Whats next? NNG -- "the next new negeration"? After that ENNG "even newer next generation"? UMNGT -- "ultimate mighty next generation thingy"?

Comment (authmillenon) [inline Makefile.dep]: The idea is to remove this prefix as soon as the new stack is stable and the old one can be removed. We did this primarily to keep colisions with the old stack as low as possible. When the network stack is done, there will be a big rename commit and `ng_` will be a thing of the past ;-)

File: sys/include/net/ng_ipv6/hdr.h [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+     * @brief Version, traffic class, and flow label
+     *
+     * @details The version are the 4 most significant bits, the traffic class
+     *          the 8 next bit, and the remainding 20 bits are the flow label
+     *
+     *         @code{.unparsed}
+     *         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     *         |Version| Traffic Class |           Flow Label                  |
+     *         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     *         @endcode
```
Comment (OlegHahm) [inline sys/include/net/ng_ipv6/hdr.h]: wouldn't it make sense to paste the complete packet structure from the RFC?

**Comment by OlegHahm** [comment]
ACK - if comment is addressed, Travis is happy and PR is squashed.

File: sys/include/net/ng_ipv6/hdr.h (hunk 2 above)
Comment (authmillenon) [inline sys/include/net/ng_ipv6/hdr.h]: I can do that

**Comment by authmillenon** [comment]
Rebased to current #2730 and squashed.

**Pull request merged**